tests/phpunit/tests/media.php

```diff
@@ -3621,6 +3621,101 @@ private function reset_omit_loading_attr_filter() {
 		// Clean up the above filter.
 		remove_filter( 'wp_omit_loading_attr_threshold', '__return_null', 100 );
 	}
+
+	/**
+	 * Test that generated files with the `image_editor_output_format` applied use the correct
+	 * quality level based on their mime type.
+	 *
+	 * @ticket 56442
+	 */
+	public function test_quality_with_image_conversion_file_sizes() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'WebP is not supported on this system.' );
+		}
+		add_filter( 'image_editor_output_format', array( $this, 'image_editor_output_jpeg' ) );
+		$temp_dir = get_temp_dir();
+		$file     = $temp_dir . '/33772.jpg';
+		copy( DIR_TESTDATA . '/images/33772.jpg', $file );
+
+		// Set JPEG output quality very low and WebP quality very high, this should force all generated WebP images to
+		// be larger than the the matching generated JPEGs.
+		add_filter( 'wp_editor_set_quality', array( $this, 'image_editor_change_quality_low_jpeg' ), 10, 2 );
+
+		$editor = wp_get_image_editor( $file );
+
+		// Verify that the selected editor actually supports WebP output.
+		if ( ! $editor->supports_mime_type( 'image/webp' ) ) {
+			$this->markTestSkipped( 'WebP is not supported by the selected image editor.' );
+		}
+
+		$attachment_id = self::factory()->attachment->create_object(
+			$file,
+			0,
+			array(
+				'post_mime_type' => 'image/jpeg',
+			)
+		);
```

Comment:
```suggestion
		$attachment_id = self::factory()->attachment->create_object(
			array(
				'post_mime_type' => 'image/jpeg',
				'file'           => $file,
			)
		);
```

You're using the deprecated argument format but if possible it would be good to use the `self::$large_id` shared fixture if possible. 

My understanding is that edits use new file names so it shouldn't be destructive but correct me if I am wrong.

Reply:
Ah, I didn't realize that, I copied the call from a nearby test. I'll commit your suggestion and verify tests still pass.